sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosClientBuilder.java

```diff
@@ -225,7 +226,7 @@ public CosmosClientBuilder resourceToken(String resourceToken) {
      * @param permissions Permission list for authentication.
      * @return current Builder.
      */
-    public CosmosClientBuilder permissions(List<Permission> permissions) {
+    public CosmosClientBuilder permissions(List<CosmosPermissionProperties> permissions) {
```

Comment:
The existing tests all are wired up against Permission (internal type) not CosmosPermissionProperties, So I I think we don't have any tests covering this API. This can be added as separate independent PR.

Reply:
Yes, I am already working on those. Will come up with a new PR for those.

Reply:
@moderakh - Here is the PR which fixes all the tests: https://github.com/Azure/azure-sdk-for-java/pull/10100